src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -129,7 +133,8 @@ public int IndexOf(T item)
         /// <param name="item">The object to insert into the <see cref="ReadOnlyCollectionBuilder{T}"/>.</param>
         public void Insert(int index, T item)
         {
-            ContractUtils.Requires(index <= _size, nameof(index));
+            if (index > _size)
```

Comment:
`List<T>` uses `(uint)index > (uint)_size` to cover the `< 0` case as well. As discussed in the description, not changing that in this PR because it's a breaking change from existing behavior. Will submit separate PR to suggest such changes.